collector/collector.go

```diff
@@ -213,7 +244,7 @@ func ScrapeTarget(ctx context.Context, target string, config *config.Module, log
 		getOids = getOids[oids:]
 	}
 
-	for _, subtree := range config.Walk {
+	for _, subtree := range newWalk {
 		var pdus []gosnmp.SnmpPDU
```

Comment:
I see, it has been done the other way here already.

Not sure about the snmp_exporter, but generally, we prefer the `pdus := []gosnmp.SnmpPDU{}` in the Prometheus codebase.

Reply:
I am not confident to be able to switch to `[]gosnmp.SnmpPDU{}` within this PR without screwing up a bunch of thing. 

I see that `ScrapeResults` struct has this type so the idea would be to update it and all code using it. It seems like there is not much of it but maybe a followup PR would be safer.

Reply:
Nothing would be screwed up, but I see now how the variable is used, which is indeed a flow with which the existing syntax has a (theoretical) performance benefit. So I retract my concern.